src/activity_sending.rs

```diff
@@ -0,0 +1,321 @@
+//! Queue for signing and sending outgoing activities with retry
+//!
+#![doc = include_str!("../docs/09_sending_activities.md")]
+
+use crate::{
+    config::Data,
+    error::Error,
+    http_signatures::sign_request,
+    reqwest_shim::ResponseExt,
+    traits::{ActivityHandler, Actor},
+    FEDERATION_CONTENT_TYPE,
+};
+use anyhow::{anyhow, Context};
+
+use bytes::Bytes;
+use futures::StreamExt;
+use http::{header::HeaderName, HeaderMap, HeaderValue};
+use httpdate::fmt_http_date;
+use itertools::Itertools;
+use openssl::pkey::{PKey, Private};
+use reqwest::Request;
+use reqwest_middleware::ClientWithMiddleware;
+use serde::Serialize;
+use std::{
+    self,
+    fmt::{Debug, Display},
+    time::{Duration, SystemTime},
+};
+use tracing::debug;
+use url::Url;
+
+/// sends an activity to many inboxes
+/// as opposed to older version, there is no queueing, retrying or parallelization.
+/// for higher performance and reliability, you'll want to use the methods in [SendActivityTask] directly
+pub async fn send_activity<Activity, Datatype, ActorType>(
+    activity: &Activity,
+    actor: &ActorType,
+    inboxes: Vec<Url>,
+    data: &Data<Datatype>,
+) -> Result<(), <Activity as ActivityHandler>::Error>
+where
+    Activity: ActivityHandler + Serialize,
+    <Activity as ActivityHandler>::Error: From<anyhow::Error> + From<serde_json::Error>,
+    Datatype: Clone,
+    ActorType: Actor,
+{
+    let sends = SendActivityTask::prepare(activity, actor, inboxes, data).await?;
+    for send in sends {
+        send.sign_and_send(data).await?;
+    }
+    Ok(())
+}
```

Comment:
This method is pretty useless, I would rather put this code with prepare and send loop in the documentation as an example for basic usage. And mention that applications can build their own queue with that.